zenoh/src/sample.rs

```diff
@@ -509,3 +518,43 @@ impl TryFrom<Reply> for Sample {
         value.sample
     }
 }
+
+/// Structure containing quality of service data
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq)]
+pub struct QoS {
```

Comment:
Using accessors, this could be a bitfield and occupy a single byte instead of 3

Reply:
What is the benefit of reducing 3  bytes to one on the system with word size >= 4 bytes ? As a separate struct it will still tend to occupy 4 bytes at least, no ?

Reply:
It's all about breakpoints and, to be fair, I think this layout just barely fits in `Sample`'s current padding, but that also means the next field is guaranteed to make `Sample` fatter, which we've shown in previous layout improvements to be rather performances sensitive for us.

API-wise, it's tempting to have only public fields, but that turns any future change to the struct's fields an API break. Builder patterns are just as convenient and give us much more flexibility in the future, both for layout optimization and for field additions.